syntax = "proto3";

package headline;

service HeadlineService {
    // Client streams batches of headlines, server returns total count when done
    rpc IngestHeadlines(stream HeadlineBatch) returns (StreamResponse);
}

message HeadlineBatch {
    repeated HeadlineRequest headlines = 1;
    int64 batch_timestamp = 2;
}

message HeadlineRequest {
    string headline = 1;
    int64 timestamp = 2;
    string symbol = 3;
    string source = 4;
}

message StreamResponse {
    int32 processed_count = 1;
    int32 batch_count = 2;
}